oci/extensions.bzl

```diff
@@ -42,16 +42,21 @@ def _oci_extension(module_ctx):
             )
 
             if mod.is_root:
-                if module_ctx.is_dev_dependency(pull):
-                    root_direct_dev_deps.append(pull.name)
-                else:
-                    root_direct_deps.append(pull.name)
+                deps = root_direct_dev_deps if module_ctx.is_dev_dependency(pull) else root_direct_deps
+                deps.append(pull.name)
+                for platform in pull.platforms:
+                    deps.append("_".join([pull.name] + platform.split("/")))
```

Comment:
Out of curiosity, what would be the reason to include these deps here? `rules_oci` v1.7 works without me including them via `use_repo` and I am curious, why I am suddenly asked to include them via the `bazel mod tidy` command.

Reply:
My understanding is it resolves an error like

```
WARNING: /builds/app/MODULE.bazel:38:20: The module extension oci defined in @rules_oci//oci:extensions.bzl reported incorrect imports of repositories via use_repo():
Not imported, but reported as direct dependencies by the extension (may cause the build to fail):
    java_base_debug_nonroot_debian12, java_base_nonroot_debian12
Imported, but reported as indirect dependencies by the extension:
    java_base_debug_nonroot_debian12_linux_amd64, java_base_debug_nonroot_debian12_linux_arm64_v8, java_base_nonroot_debian12_linux_amd64, java_base_nonroot_debian12_linux_arm64_v8
Fix the use_repo calls by running 'bazel mod tidy'.
```

Reply:
`bazel mod tidy` could support something like `absolutely neccesary repos` vs `could have if build requires it` kind of `use_repo` calls.

Reply:
but this what we have to do for now, if we want to avoid that annoying warning message. @fmeum probably has better ideas for this.

Reply:
I like @thesayyn's idea of something similar to a `root_module_optional_deps` attribute that silences the warning without also adding `use_repo` arguments. Could you file an issue for this if you think it would be useful?